app/models/account.rb

```diff
@@ -112,6 +112,11 @@ class Account < ApplicationRecord
 
   delegate :chosen_languages, to: :user, prefix: false, allow_nil: true
 
+  # Convert newlines so validation doesn't count them as two characters
+  def note=(val)
+    self['note'] = val.gsub("\r\n", "\n")
```

Comment:
I suggest:

```ruby
before_validation :normalize_note

private

def normalize_note
  self.note = note.gsub("\r\n", "\n")
end
```

Worth noting I do not remember something like this for status text...

Reply:
This code is used for counting a status text:

https://github.com/tootsuite/mastodon/blob/bf70e5cfdaef9b7a3a927548a05203a864d5bea9/app/validators/status_length_validator.rb#L18

I couldn’t make a test triggering `before_validation`, so I didn’t go for you suggestion or for using `grapheme_length`.